src/AbstractArray.php

```diff
@@ -638,4 +638,23 @@ protected function sortingKeys(array &$elements, $order = SORT_ASC, $strategy =
                 ksort($elements, $strategy);
         }
     }
+
+    /**
+     * Returns the first value that matches $func condition
+     *
+     * @param Closure $func
+     *
+     * @return mixed The resulting value
+     */
+    public function find(Closure $func)
+    {
+        foreach ($this->elements as $key => $value) {
+            $current = call_user_func($func, $value, $key, $this->elements);
```

Comment:
I think we should replace `call_user_func` with `$func($value, $key)`. We'll get a bit performance improvement, check my [performance test](https://ideone.com/VQ3Hwm)